src/schemas/user.js

```diff
@@ -12,6 +22,36 @@ const userSchema = new mongoose.Schema({
     required: true,
     unique: true
   },
+  username: {
+    type: String,
+    required: true
+  },
+  skills: {
+    type: [Schema.Types.ObjectId], // department
+    default: []
+  },
+  knowledge: {
+    type: [Schema.Types.ObjectId], // department
+    default: []
+  },
```

Comment:
I didn't quite catch why `skills` and `knowledge` is using `ObjectID` if those are not represented by a model

Reply:
Also, remove the unwanted comments